### PR TITLE
self destruct exterminators on objective completion

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

Adds a system to blow up exterminators 60 seconds after they kill their target.

## Why / Balance

Exterminators have the sole goal of killing someone then "shutting off" but the issue is that the lack of mechanical reinforcement really just results in exterminators running around as nigh-invincible murder hobos even after their target is dead. To lessen admin burden and just to mechanically reinforce the intended gameplay, terminators are now given a short self-destruct sequence that makes sure that they are gone once they finish their job.

removes the objective to self destruct as it is now pretty much implicit.

## Media

image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Exterminators now enter a 60 second self destruct sequence upon terminating their target.

### CODE REVIEW

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by Hanzdegloker** [comment]
finally, that one round where the exterminator teamed up with the nukies won't happen again.

**Comment by thebadman4662** [comment]
Is the explosion enough to gib target? Cause if they get revived you will both self destruct and fail to kill.

**Comment by SoulFN** [comment]
> Is the explosion enough to gib target? Cause if they get revived you will both self destruct and fail to kill.

It`s explodes like microbomb implant

**Comment by SlamBamActionman** [comment]
Is there any check to determine that the Terminator is indeed responsible for the target's death? I'd hate to explode on the other side of the station just because my target touched a live wire, only for them to get dragged to medbay and revived 5 seconds later.

**Comment by deltanedas** [comment]
> This was explicitly stated as a viable option for the Terminator in the Terminator design doc

thats worded that you can help nukies if your target is alive but after they die you have no business helping them

**Comment by KingOnBudget** [comment]
This should activate after the target is gibbed as for the greentext exterminator target has be unrecoverable. Trying to gib your target while the entire station is trying to kill you and drag your target away is already hard as it is. Giving a timer to that will just make the objective tedious imo.

**Comment by SlamBamActionman** [comment]
> This should activate after the target is gibbed as for the greentext exterminator target has be unrecoverable. Trying to gib your target while the entire station is trying to kill you and drag your target away is already hard as it is. Giving a timer to that will just make the objective tedious imo.

This would go well with the suggestion from this Discord thread: 
https://discord.com/channels/310555209753690112/1220389684866187344/1220800475792932874

Make the bomb an Action available to the Terminator that they are able to activate at will (to give Terminators some control over the explosion), but if the target is gibbed the bomb starts automatically.

**Comment by Tayrtahn** [comment]
Hold on, is the exterminator intended to round remove the target or just kill them?

I know RR makes more sense with the (implied) lore, but which is more fun for the game?

**Comment by deltanedas** [comment]
RR for the sweet sweet greentext

**Comment by FairlySadPanda** [comment]
If you don't have the robot represent the threat of RR, it can be metagamed incredibly easily by just walking to medbay, getting beaten to death, then being revived. Especially in the current medical situation where there's no punishment for it except med getting annoyed at you for wasting a few chems.

**Comment by EmoGarbage404** [comment]
> This should activate after the target is gibbed as for the greentext exterminator target has be unrecoverable. Trying to gib your target while the entire station is trying to kill you and drag your target away is already hard as it is. Giving a timer to that will just make the objective tedious imo.

it activates when all the objectives are complete. the objective is marked green when the person dies, not when they're unrecoverable. if that's the intended gameplay than the objective should be changed to reflect that.

File: Content.Server/Terminator/Systems/TerminatorSystem.cs [hunk 1]
```diff
@@ -63,4 +80,38 @@ public EntityUid CreateSpawner(EntityCoordinates coords, EntityUid? target)
...
+                    _explosion.TriggerExplosive(terminator);
+                }
+
+                continue;
+            }
+
+            if (!_objectives.AllObjectivesComplete((uid, mindComp)))
+                continue;
+
+            if (mindComp.Session != null)
```
Comment (metalgearsloth) [inline Content.Server/Terminator/Systems/TerminatorSystem.cs:108]: Adding an expensive update check every tick is pretty ass, does adding a target objective component not work or smth.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Agree with delta's review as it's less snowflake data terminator requires.

File: Content.Server/Terminator/Systems/TerminatorSystem.cs (hunk 1 above)
Comment (deltanedas) [reply Content.Server/Terminator/Systems/TerminatorSystem.cs:108]: oh yeah this could be on the objective entity

**Pull request closed without merging**